# Cleanning of Dataset

1. Getting rid irrelevant header information and footnotes
2. Change dtype of numerical values from type objects into int
3. Change dtype of date column into from type object into datetime
4. Making a dataset specific to the time period from when covid started to present

Dataset used: 
1. Formation Of Business Entities, By Industry, Monthly
2. Cessation Of Business Entities, By Industry, Monthly

In [24]:
# Import libraries

import pandas as pd
# import seaborn as sb
# import numpy as np
# import matplotlib.pyplot as plt

In [25]:
# Reading the csv files

formation = pd.read_csv('FormationData.csv')

cessation = pd.read_csv('CessationData.csv')

In [26]:
# removing irelevalent header information and footnotes
# formation dataset

holder = formation.drop(formation.index[0:9], axis=0)
formation = holder.drop(formation.index[26:49], axis=0)


In [27]:
# Orientating the dataset and setting the header
formation = formation.T

formation = formation.reset_index(drop=True)

header = formation.iloc[0]

formation = formation[1:]

formation.columns = header

#formation

In [28]:
#removing whitespaces from the column headers

formation.columns = formation.columns.str.strip()

print(formation.columns)


Index(['Data Series', 'Total', 'Manufacturing', 'Construction',
       'Wholesale Trade', 'Retail Trade', 'Transportation & Storage',
       'Accommodation', 'Food & Beverage Services',
       'Information & Communications', 'Finance & Insurance', 'Real Estate',
       'Professional Services', 'Administrative & Support Services',
       'Education, Health & Social Services',
       'Arts, Entertainment, Recreation & Other Services', 'Others'],
      dtype='object', name=0)


In [29]:
# Changing the column (industry) values into numeric

formation.iloc[:, 1:] = formation.iloc[:, 1:].apply(pd.to_numeric)

In [30]:
# Changing the colume (Data Series) value into date time

formation['Data Series'] = formation['Data Series'].str.strip()
formation['Data Series'] = pd.to_datetime(formation['Data Series'], format='%Y %b')


In [31]:
# Checking conversion
formation['Data Series']

1     2022-12-01
2     2022-11-01
3     2022-10-01
4     2022-09-01
5     2022-08-01
         ...    
392   1990-05-01
393   1990-04-01
394   1990-03-01
395   1990-02-01
396   1990-01-01
Name: Data Series, Length: 396, dtype: datetime64[ns]

In [32]:
#Save dataset 
formation.to_csv('Formation_Cleaned.csv', index=False)


In [33]:
# Filter time period from when covid-19 first started 
covid_formation = formation[formation['Data Series'] >= "2020-01-01"]
# covid_formation.reset_index(inplace=True)

# Save the dataset

covid_formation.to_csv('Covid_Formation.csv', index=False)

In [34]:
#Create dataset for pre-covid

pre_covid_formation = formation[formation['Data Series'] < "2020-01-01"]
# pre_covid_formation.reset_index(inplace=True)

# Save the dataset
pre_covid_formation.to_csv('Pre_covid_Formation.csv', index=False)


In [35]:
pre_covid_formation

0   Data Series Total Manufacturing Construction Wholesale Trade Retail Trade  \
37   2019-12-01  4715           153          165             891          432   
38   2019-11-01  5182           167          186             937          455   
39   2019-10-01  5433           179          215             993          520   
40   2019-09-01  5549           189          190             934          527   
41   2019-08-01  5126           185          176             875          460   
..          ...   ...           ...          ...             ...          ...   
392  1990-05-01  2276           237          233             630          325   
393  1990-04-01  1958           175          202             627          264   
394  1990-03-01  2375           248          239             668          325   
395  1990-02-01  1997           212          216             548          261   
396  1990-01-01  1719           168          147             536          207   

0   Transportation & Storage Accommodation Food & Beverage Services  \
37                       249            10                      181   
38                       252             5                      241   
39                       237             7                      253   
40                       302             6                      282   
41                       260             6                      261   
..                       ...           ...                      ...   
392                      109             2                       84   
393                      104             1                       67   
394                      126             1                       83   
395                      112             2                       74   
396                      119             4                       52   

0   Information & Communications Finance & Insurance Real Estate  \
37                           507                 589          47   
38                           623                 679          47   
39                           626                 564          67   
40                           645                 590          63   
41                           594                 570          72   
..                           ...                 ...         ...   
392                           47                  73          86   
393                           39                  70          62   
394                           61                  82          76   
395                           49                  57          69   
396                           24                  51          66   

0   Professional Services Administrative & Support Services  \
37                    795                               209   
38                    810                               208   
39                    891                               236   
40                    969                               224   
41                    875                               229   
..                    ...                               ...   
392                   170                                97   
393                   132                                65   
394                   166                               121   
395                   143                               113   
396                   124                                89   

0   Education, Health & Social Services  \
37                                  212   
38                                  273   
39                                  275   
40                                  288   
41                                  247   
..                                  ...   
392                                  41   
393                                  32   
394                                  40   
395                                  37   
396                                  34   

0   Arts, Entertainment, Recreation & Other Services Others  
37                   

In [36]:
# removing irrelevent header information and footnotes
# cessation dataset

holder = cessation.drop(cessation.index[0:9], axis=0)
cessation = holder.drop(cessation.index[26:49], axis=0)

cessation

Unnamed: 0 Unnamed: 1 Unnamed: 2  \
9                                         Data Series  2022 Dec   2022 Nov    
10                                              Total       3856       4496   
11                                      Manufacturing        186        168   
12                                       Construction        153        209   
13                                    Wholesale Trade        654        818   
14                                       Retail Trade        496        528   
15                           Transportation & Storage        215        253   
16                                      Accommodation          5         10   
17                           Food & Beverage Services        209        275   
18                       Information & Communications        403        464   
19                                Finance & Insurance        289        293   
20                                        Real Estate         47         44   
21                              Professional Services        604        685   
22                  Administrative & Support Services        179        223   
23                Education, Health & Social Services        194        219   
24    Arts, Entertainment, Recreation & Other Serv...        203        271   
25                                             Others         19         36   

   Unnamed: 3 Unnamed: 4 Unnamed: 5 Unnamed: 6 Unnamed: 7 Unnamed: 8  \
9   2022 Oct   2022 Sep   2022 Aug   2022 Jul   2022 Jun   2022 May    
10       4418       4503       4258       4389       3998       4242   
11        184        185        186        190        200        171   
12        202        232        213        208        165        247   
13        743        828        764        826        720        769   
14        553        516        524        469        513        477   
15        281        280        260        286        295        280   
16          2         11          4         13          4          8   
17        284        246        245        213        172        233   
18        429        455        479        426        426        415   
19        355        294        240        305        225        282   
20         51         47         53         46         49         49   
21        641        678        619        687        565        673   
22        194        228        195        203        160        203   
23        224        195        220        214        217        193   
24        241        275        237        284        264        215   
25         34         33         19         19         23         27   

   Unnamed: 9  ... Unnamed: 387 Unnamed: 388 Unnamed: 389 Unnamed: 390  \
9   2022 Apr   ...    1990 Oct     1990 Sep     1990 Aug     1990 Jul    
10       4468  ...         1103         1038         1148         1509   
11        221  ...          101           97          141          150   
12        197  ...          107           95          103          166   
13        765  ...          310          294          335          396   
14        547  ...          231          194          187          291   
15        280  ...           47           50           54           59   
16          4  ...            0            2            1            2   
17        245  ...           36           50           42           78   
18        424  ...           17           19           17           21   
19        305  ...           30           25           19           32   
20         47  ...           26           32           33           38   
21        685  ...           52           45           60           69   
22        229  ...           56           39           46           64   
23        233  ...           15           16           20           32   
24        262  ...           66           69           84          100   
25         24  ...            9           11            6        

In [37]:
# Orientating the dataset and setting the header
cessation = cessation.T

cessation = cessation.reset_index(drop=True)

header = cessation.iloc[0]

cessation = cessation[1:]

cessation.columns = header

#cessation

In [38]:
#removing whitespaces from the column headers

cessation.columns = cessation.columns.str.strip()

print(cessation.columns)

Index(['Data Series', 'Total', 'Manufacturing', 'Construction',
       'Wholesale Trade', 'Retail Trade', 'Transportation & Storage',
       'Accommodation', 'Food & Beverage Services',
       'Information & Communications', 'Finance & Insurance', 'Real Estate',
       'Professional Services', 'Administrative & Support Services',
       'Education, Health & Social Services',
       'Arts, Entertainment, Recreation & Other Services', 'Others'],
      dtype='object', name=0)


In [39]:
# Changing the column (industry) values into numeric

cessation.iloc[:, 1:] = cessation.iloc[:, 1:].apply(pd.to_numeric)

In [40]:
# Changing the colume (Data Series) value into date time

cessation['Data Series'] = cessation['Data Series'].str.strip()
cessation['Data Series'] = pd.to_datetime(cessation['Data Series'], format='%Y %b')

In [41]:
# Checking conversion
cessation['Data Series']

1     2022-12-01
2     2022-11-01
3     2022-10-01
4     2022-09-01
5     2022-08-01
         ...    
392   1990-05-01
393   1990-04-01
394   1990-03-01
395   1990-02-01
396   1990-01-01
Name: Data Series, Length: 396, dtype: datetime64[ns]

In [42]:
#Save dataset 
cessation.to_csv('Cessation_Cleaned.csv', index=False)

In [43]:
# Filter time period from when covid-19 first started 
covid_cessation = cessation[cessation['Data Series'] >= "2020-01-01"]
#covid_cessation.reset_index(inplace=True)

# Save the dataset

covid_cessation.to_csv('Covid_Cessation.csv', index=False)

In [44]:
#Create dataset for pre-covid

pre_covid_cessation = cessation[cessation['Data Series'] < "2020-01-01"]
#pre_covid_cessation.reset_index(inplace=True)

# Save the dataset

pre_covid_cessation.to_csv('Pre_covid_Cessation.csv', index=False)

In [45]:
covid_formation['Data Series']

1    2022-12-01
2    2022-11-01
3    2022-10-01
4    2022-09-01
5    2022-08-01
6    2022-07-01
7    2022-06-01
8    2022-05-01
9    2022-04-01
10   2022-03-01
11   2022-02-01
12   2022-01-01
13   2021-12-01
14   2021-11-01
15   2021-10-01
16   2021-09-01
17   2021-08-01
18   2021-07-01
19   2021-06-01
20   2021-05-01
21   2021-04-01
22   2021-03-01
23   2021-02-01
24   2021-01-01
25   2020-12-01
26   2020-11-01
27   2020-10-01
28   2020-09-01
29   2020-08-01
30   2020-07-01
31   2020-06-01
32   2020-05-01
33   2020-04-01
34   2020-03-01
35   2020-02-01
36   2020-01-01
Name: Data Series, dtype: datetime64[ns]

In [46]:
#creating a dataset to store the values for show the current number of business entities, by industry

total = formation.sub(cessation)

total

0   Data Series Total Manufacturing Construction Wholesale Trade Retail Trade  \
1        0 days  1279            11           94             150          -74   
2        0 days  1165            64           57              83         -118   
3        0 days   694             8           49              86         -148   
4        0 days  1166             7           58              27          -61   
5        0 days  1601            30           78             148           -1   
..          ...   ...           ...          ...             ...          ...   
392      0 days   982           108           96             264          109   
393      0 days   723            29           76             307           47   
394      0 days   975            96           84             272          109   
395      0 days   852            95           89             225           51   
396      0 days   422            26           -9             193           -9   

0   Transportation & Storage Accommodation Food & Beverage Services  \
1                        -43            -2                        5   
2                        -37            -4                      -42   
3                        -63             1                        5   
4                        -54            -3                       76   
5                        -17            15                       50   
..                       ...           ...                      ...   
392                       48             1                       19   
393                       31             0                       -1   
394                       62             0                       17   
395                       64             1                       31   
396                       63             2                        1   

0   Information & Communications Finance & Insurance Real Estate  \
1                            290                 389           9   
2                            313                 410          16   
3                            262                 237          24   
4                            288                 429          33   
5                            377                 463           6   
..                           ...                 ...         ...   
392                           29                  52          57   
393                           29                  57          21   
394                           42                  62          43   
395                           33                  33          46   
396                            4                  24          38   

0   Professional Services Administrative & Support Services  \
1                     275                                17   
2                     298                               -11   
3                     139                                 5   
4                     231                                 0   
5                     278                                14   
..                    ...                               ...   
392                   101                                42   
393                    73                                11   
394                    91                                45   
395                    91                                52   
396                    54                                20   

0   Education, Health & Social Services  \
1                                    63   
2                                    77   
3                                    14   
4                                    87   
5                                     7   
..                                  ...   
392                                  21   
393                                  15   
394                                  19   
395                                  13   
396                                  10   

0   Arts, Entertainment, Recreation & Other Services Others  
1                    

# After Cleaning 

3 Datasets for formation
- One dataset consisting of all formations of business entities, by industry, monthly
- One dataset consisting of all formations of business entities, by industry, monthly, after Covid-19 struck Singapore
- One dataset consisting of all formations of business entities, by industry, monthly, precovid

3 Datasets for cessation
- One dataset consisting of all cessation of business entities, by industry, monthly
- One dataset consisting of all cessation of business entities, by industry, monthly, after Covid-19 struck Singapore
- One dataset consisting of all cessation of business entities, by industry, monthly, precovid
<br> </br>

<li> A total of 15 industries excluding total </li>
<li> A total of 394 months from 1990 Jan to 2022 Nov
<li> A total of 35 months after covid struck from 2020 Jan to 2022 Nov
<li> A total of 359 months of precovid data from 1990 Jan to 2019 Dec